### PR TITLE
[riscocloud][WIP] Risco Cloud Binding initial contribution

### PR DESCRIPTION
**Pull request by scantineau**
Binding to Risco Cloud

This binding connects to www.riscocloud.com using provided credentials.

It allows you to monitor your secured sites and to arm/disarm them.

After adding this binding, you have to create a new thing of type "Main bridge" and to provide credentials.

When "Main bridge" is ONLINE, use the discovery feature or go to add thing menu and the binding will retrieve connected sites and add them to your INBOX.

When added and ONLINE, once again, use the discovery feature or go to add thing menu and the binding will create an overview thing and one or more part thing(s) for each connected sites.

This binding is in pre-release so feel free to report any bugs

### CODE REVIEW

**Comment by scantineau** [comment]
Hello @wborn 

Sorry I didn't undestood the signed-off had to be in the commit message.

I think we cannot change this comment so how to proceed ? 

Thanks

**Comment by wborn** [comment]
Thanks for the binding @scantineau! That's right, each commit should contain the signed off by line.

Since you only have one commit it should be relatively easy to "amend" your commit message, see: https://help.github.com/articles/changing-a-commit-message/

**Comment by cweitkamp** [comment]
AFAIK you have to add it to the parent [pom.xml](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/pom.xml). Please note that it is in alphabetical order.

**Comment by scantineau** [comment]
Thanks to @cweitkamp !

Does anyone knows why Jenkins still wants to use the old bundle name ? 

`08:30:28.673  |  [ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:1.2.0:validate-id (default-validate-id) on project org.openhab.binding.myelas: The Maven artifactId (currently: "org.openhab.binding.myelas") must be the same as the bundle symbolic name (currently: "org.openhab.binding.riscocloud") -> [Help 1]`

File: addons/binding/org.openhab.binding.riscocloud/build.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+source..=src/main/java/
+output..=target/classes
+bin.includes = META-INF/,\
+               .,\
+               OSGI-INF/,\
+               ESH-INF/,\
+               NOTICE,\
```
Comment (wborn) [inline addons/binding/org.openhab.binding.riscocloud/build.properties]: To fix the build change `NOTICE` to `about.html`

**Comment by cweitkamp** [comment]
The next step is to get rid of the errors reported by our Code Analysis Tool (e.g. license headers and `@author` tags). Have a look at the [coding guidelines A.2 and A.4](https://www.openhab.org/docs/developer/development/guidelines.html#a-code-style) for more information.

```
[ERROR] Code Analysis Tool has found: 
 20 error(s)! 
 9 warning(s) 
 14 info(s)
[WARNING] .binding.riscocloud/ESH-INF/thing/overview.xml:[8]
There were whitespace characters used for indentation. Please use tab characters instead
[WARNING] .binding.riscocloud/ESH-INF/thing/siteBridge.xml:[7]
There were whitespace characters used for indentation. Please use tab characters instead
[WARNING] .binding.riscocloud/ESH-INF/thing/mainBridge.xml:[7]
There were whitespace characters used for indentation. Please use tab characters instead
[WARNING] .binding.riscocloud/ESH-INF/thing/part.xml:[8]
There were whitespace characters used for indentation. Please use tab characters instead
[ERROR] .binding.riscocloud/META-INF/MANIFEST.MF:[0]
The MANIFEST.MF file must not contain any Require-Bundle entries. Instead, Import-Package must be used.
[ERROR] .binding.riscocloud/META-INF/MANIFEST.MF:[0]
The package org.openhab.binding.riscocloud should be marked as "internal" if it is not exported.
[ERROR] .binding.riscocloud/META-INF/MANIFEST.MF:[0]
The package org.openhab.binding.riscocloud.handler should be marked as "internal" if it is not exported.
[ERROR] .binding.riscocloud/META-INF/MANIFEST.MF:[0]
The package org.openhab.binding.riscocloud.json should be marked as "internal" if it is not exported.
[WARNING] .binding.riscocloud/META-INF/MANIFEST.MF:[13]
The version of the package com.google.gson should not be specified
[ERROR] org.openhab.binding.riscocloud.RiscoCloudBindingConstants.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.handler.SiteHandler.java:[94]
Avoid printStackTrace(); use a logger call instead.
[WARNING] org.openhab.binding.riscocloud.handler.SiteHandler.java:[112]
Avoid throwing raw exception types.
[ERROR] org.openhab.binding.riscocloud.handler.WebSiteInterface.java:[1]
Header is missing
[ERROR] org.openhab.binding.riscocloud.handler.WebSiteInterface.java:[30]
An author tag is missing
[WARNING] org.openhab.binding.riscocloud.handler.WebSiteInterface.java:[30]
Missing Javadoc comment.
[ERROR] org.openhab.binding.riscocloud.handler.LoginResult.java:[1]
Header is missing
[ERROR] org.openhab.binding.riscocloud.handler.LoginResult.java:[8]
An author tag is missing
[WARNING] org.openhab.binding.riscocloud.handler.LoginResult.java:[8]
Missing Javadoc comment.
[ERROR] org.openhab.binding.riscocloud.internal.RiscoCloudHandlerFactory.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.Part.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.Detector.java:[1]
Header is missing
[ERROR] org.openhab.binding.riscocloud.json.PartInfo.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.Detectors.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.AllGrpState.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.ServerDatasObject.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.LastAlarm.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.ServerDatasHandler.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.riscocloud.json.Overview.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[WARNING] .binding.riscocloud/ESH-INF/thing/mainBridge.xml:[0]
Unused bridge reference with id - riscoCloudServer
```

**Comment by scantineau** [comment]
Ok thanks !

I also have to fix some minor issues.

Maybe we should label WIP ?

**Comment by cweitkamp** [comment]
> Maybe we should label WIP

Feel free to add [WIP] to the title of the PR if you are not confident with is so far. We will take care of the rest.

**Comment by davidgraeff** [comment]
Half a year no progress. Is this still being worked on?

**Comment by scantineau** [comment]
I don't have enough time to work on it for the moment.

I hope I will have some in the next months

**Comment by pergolafabio** [comment]
@scantineau 

what method are you using to get the state arm/disarmed? do you poll www.riscocloud.com ?or do you have an API? or how does your code work?

i am now using this component  https://github.com/lucacalcaterra/risco-mqtt-bridge
working verry well, only 1 issue , is the state

when you arm/disarm from a risco hardware panel, you see the state "overview"  changing after 4-5 minutes on www.riscocloud.com , this delay if then offcourse also reflected to the polling

i know risco has an official API, but in order to get it , we need a business case

**Comment by scantineau** [comment]
I analyzed how the site works and I use the same requests.

www.riscocloud.com polls his own api every 5 seconds.

Check your polling interval. Luca uses the same api as me.

**Comment by pergolafabio** [comment]
@scantineau 
yes , had contact with Luca, he confirmed the delay issue, he also had contact with risco, there is nothing he can do about it
you can check it too  if you want

just arm/disarm from a panel (not app), then look at www.riscocloud.com , look at the overview page, the one with the slider that goes to right/lef
you will see that slider changing after about 4-5 minutes.... that delay is also present in luca his app/code

he is going to change the code, not based on that slider/overview page, but based on an event, seems that is instant coming in.....

**Comment by Hilbrand** [comment]
@scantineau Do you have plans to finish work on this binding in the short term? Otherwise we'll close it for now and you can open a new pr in the future.

**Comment by scantineau** [comment]
Work is in progress again (sorry for the time it took).
Actual state is a stable connection with Risco Cloud service (connect, reconnect with multiple tries...).
Only an overview is available (number of armed/disarmed/partiallyArmed) in read only.
Next step is to send command to Risco Cloud (arm/disarm/partiallyArm).

**Comment by pergolafabio** [comment]
if you wait like 2-3 months, an API will be available...

**Comment by scantineau** [comment]
We will see if there will be so many changes. 
I'm already using a REST api... I only had to parse a single webpage for the site list.

**Comment by pergolafabio** [comment]
Are you using code based on webpage or from the risco app?

**Comment by scantineau** [comment]
From webpage. I don't have the app source code.

**Comment by pergolafabio** [comment]
Have a look at this one... I am using that with home assistant, it's the best project I have found, I can use now the motion created by the dects...

He sniffed the code from the app...

https://github.com/mancioshell/node-risco-client , it's much faster... ,Also in getting states

**Comment by scantineau** [comment]
Urls are different but same mechanic.
I will have a try if it's faster but it's not slow that much at the moment...
By the way, there is no push method for events, so there always be a lag between the real state and when you will get it (because the binding have to poll)

**Comment by pergolafabio** [comment]
It's still poll indeed, but the polling from the other url from app are faster to get the state of the dects... For me , it almost changes instant... I use them now to turn on lights in rooms when I walk before them

**Comment by scantineau** [comment]
what's your polling value ?

**Comment by pergolafabio** [comment]
1500 ms

**Comment by Hilbrand** [comment]
I'm closing this pr for now. There have been several stuctural changes since this pr and some more expected with openHAB 3.0. Therefor it's better to open a new pr in the future. If you need any help migrating or updating feel free to ask for help.

**Pull request closed without merging**